processor/deltatocumulativeprocessor/internal/metrics/data.go

```diff
@@ -0,0 +1,56 @@
+package metrics
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/data"
+)
+
+type Data[D data.Point[D]] interface {
+	At(i int) D
+	Len() int
+	Ident() Ident
+}
+
+type Sum Metric
+
+func (s Sum) At(i int) data.Number {
+	dp := Metric(s).Sum().DataPoints().At(i)
+	return data.Number{NumberDataPoint: dp}
+}
+
+func (s Sum) Len() int {
+	return Metric(s).Sum().DataPoints().Len()
+}
+
+func (s Sum) Ident() Ident {
+	return (*Metric)(&s).Identity()
```

Comment:
Is casting necessary here?

Reply:
same as above, the method is defined on `*Metric`, not `Sum`